src/spikeinterface/core/tests/test_analyzer_extension_core.py

```diff
@@ -218,7 +218,7 @@ def test_delete_on_recompute(create_cache_folder):
     sorting_analyzer.compute("templates")
 
     # re compute random_spikes should delete waveforms and templates
-    sorting_analyzer.compute("random_spikes")
+    sorting_analyzer.compute("random_spikes", force_recompute=True)
     assert sorting_analyzer.get_extension("templates") is None
     assert sorting_analyzer.get_extension("waveforms") is None
 
```

Comment:
```suggestion
def test_delete_on_parm_change(create_cache_folder):
    cache_folder = create_cache_folder
    sorting_analyzer = get_sorting_analyzer(cache_folder, format="memory", sparse=False)

    sorting_analyzer.compute("random_spikes", seed=1, force_recompute=False)
    sorting_analyzer.compute("waveforms")
    sorting_analyzer.compute("templates")

    sorting_analyzer.compute("random_spikes", seed=1, force_recompute=False)
    assert sorting_analyzer.get_extension("waveforms") is not None

    sorting_analyzer.compute("random_spikes", seed=2, force_recompute=False)
    assert sorting_analyzer.get_extension("waveforms") is None
    assert sorting_analyzer.get_extension("templates") is None
```

I'd be up for another test here, just to make sure that the `random_spikes` parameter change doesn't delete waveforms when a parameter isn't changed, but does delete it when it is.

Reply:
typically for me random_spike recompute with the same number of spikes and seed=None (so same params) should delete everything but I guess it will not be the case with this PR

Reply:
Yup, for this PR it wouldn't recompute or delete with seed=None, you'd have to turn on `force_recompute=True`. I do think `random_spikes` feels special for this kinda thing, compared to the other extensions. But I'm not sure what the solution should be.